### PR TITLE
Fix enum flag from cpp conversion

### PR DESCRIPTION
**Pull request by skallweitNV**
- Use `__str__` from the underlying flag factory class (allows printing flags like `Flag.A|B|C`)
- Fix conversion from cpp -> python of enum flags that contain a combination of flags.

### CODE REVIEW

**Comment by skallweitNV** [comment]
Hmm, looks like `__str__` has changed in Python >= 3.11. It now returns numeric values for the `Int` variants.

File: src/nb_enum.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ PyObject *enum_create(enum_init_data *ed) noexcept {
     scope.attr(name) = result;
     result.attr("__doc__") = ed->docstr ? str(ed->docstr) : none();
 
-    result.attr("__str__") = enum_mod.attr("Enum").attr("__str__");
```
Comment (wjakob) [inline src/nb_enum.cpp:74]: Can you use `factor_name` (and otherwise `Enum`) only when this is a flag?

**Review by wjakob — COMMENTED** [review]
Could you add a testcase that checks the behavior of the flag `__str__` routine?

File: src/nb_enum.cpp (hunk 1 above)
Comment (skallweitNV) [reply src/nb_enum.cpp:74]: Done.

**Comment by skallweitNV** [comment]
@wjakob this should be good to merge now.

**Pull request merged**